packages/trpc/server/routers/loggedInViewer/deleteCredential.handler.ts

```diff
@@ -328,52 +327,30 @@ export const deleteCredentialHandler = async ({ ctx, input }: DeleteCredentialOp
     }
   }
 
-  // Validated that credential is user's above
-  await prisma.credential.delete({
-    where: {
-      id: id,
-    },
-  });
-
   // Backwards compatibility. Selected calendars cascade on delete when deleting a credential
   // If it's a calendar remove it from the SelectedCalendars
   if (credential.app?.categories.includes(AppCategories.calendar)) {
-    const selectedCalendars = await prisma.selectedCalendar.findMany({
+    const calendar = await getCalendar(credential);
+
+    const calendars = await calendar?.listCalendars();
+
+    const calendarIds = calendars?.map((cal) => cal.externalId);
+
+    await prisma.selectedCalendar.deleteMany({
       where: {
         userId: user.id,
         integration: credential.type as string,
+        externalId: {
+          in: calendarIds,
+        },
       },
     });
-
-    if (selectedCalendars.length) {
-      const calendar = await getCalendar(credential);
-
-      const calendars = await calendar?.listCalendars();
-
-      if (calendars && calendars.length > 0) {
-        calendars.map(async (cal) => {
-          prisma.selectedCalendar
-            .delete({
-              where: {
-                userId_integration_externalId: {
-                  userId: user.id,
-                  externalId: cal.externalId,
-                  integration: cal.integration as string,
-                },
-              },
-            })
-            .catch((error) => {
-              if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === "P2025") {
-                console.log(
-                  `Error deleting selected calendars for user ${user.id} and calendar ${credential.appId}. Could not find selected calendar.`
-                );
-              }
-              console.log(
-                `Error deleting selected calendars for user ${user.id} and calendar ${credential.appId} with error: ${error}`
-              );
-            });
-        });
-      }
```

Comment:
Because of the above reason, if these credentials were deleted on cascade, this would throw an error.